### ISSUE TITLE
Decomposition then ligaturing seems to break mark to ligature positioning

### ISSUE THREAD

**Issue by khaledhosny** [harfbuzz/harfbuzz#4969]
I’m working on a font where some glyphs are decomposed to parts, and mark to base positioning works fine. But some of the decomposed glyphs form ligatures, and this seems to make mark to ligature positioning fail.

In this test font, final lam is decomposed into two parts, then beh makes a ligature with the first part. The mark over the lam gets mispostioned (looks like the anchor positioning is not applied):

[NewFont-Regular.zip](https://github.com/user-attachments/files/18068925/NewFont-Regular.zip)

```
$ hb-shape NewFont-Regular.ttf -u 066E,064E,0644,064E
[fatha-ar=0+0|_lam.bowel=0+537|fatha-ar=0@20,370+0|behDotless_lam-ar.init=0+246]
```
![bug](https://github.com/user-attachments/assets/e13f88e3-e15e-46e4-85a8-7799c5bcd026)

If the ligature formation is disabled, the mark is correctly positioned:

```
$ hb-shape NewFont-Regular.ttf -u 066E,064E,0644,064E --features=-liga
[fatha-ar=2@369,717+0|_lam.bowel=2+537|_lam=2+108|fatha-ar=0@20,370+0|behDotless-ar.init=0+220]
```
![noliga](https://github.com/user-attachments/assets/d5072c2a-d824-4f4d-896f-d6483e60cdf3)

DirectWrite seems to get the mark positioning correct even with ligature:

```
$ hb-shape.exe NewFont-Regular.ttf -u 066E,064E,0644,064E --shaper=directwrite
[fatha-ar=3@359,727+0|_lam.bowel=2+537|fatha-ar=1@20,370+0|behDotless_lam-ar.init=0+246]
```
![correct](https://github.com/user-attachments/assets/112bc3b5-dd61-41dc-8703-28b5d0fa112d)

CoreText has the same issue as HarfBuzz.

**Comment by behdad** [comment]
cc @nedley for CoreText.

That part of the code is so complicated, I need to see what to do to match DWrite without breaking other things...

**Comment by behdad** [comment]
So walk me through this a bit. You expect the mark to try to attach to the first glyph in a multiple-subst, yeah? That's something we do for mark-to-base already, so it makes sense to do for mark-ligature as well. Lemme work on that.

**Comment by behdad** [comment]
> So walk me through this a bit. You expect the mark to try to attach to the first glyph in a multiple-subst, yeah? That's something we do for mark-to-base already, so it makes sense to do for mark-ligature as well. Lemme work on that.

Here's the mark-to-base logic for that:
https://github.com/harfbuzz/harfbuzz/blob/c8050b3872d012adf0a64b5b39d9c67e7f4dfc94/src/OT/Layout/GPOS/MarkBasePosFormat1.hh#L93-L110

**Comment by behdad** [comment]
> Here's the mark-to-base logic for that:

So I added that same logic to mark-to-ligature, but it still doesn't work, because: the ligation happens *after* the multiplication. So the second part of the Lam *doesn't* get the ligature-id of the ligature.

We can mitigate that by: always setting the lig-id of the subsequent parts of a multiple subst when a ligature forms. I'll see about that.

**Comment by khaledhosny** [comment]
> You expect the mark to try to attach to the first glyph in a multiple-subst, yeah?

Yes, exactly. As it happens when there is no ligaturing (which was fixed long ago to match Uniscribe as well).
